Add comprehensive fve-dd documentation with examples for BitLocker and LUKS unlocking (#94)
fox-it/dissect-docs: docs/source/projects/dissect.fve/index.rst

dissect.fve
===========

.. button-link:: https://github.com/fox-it/dissect.fve
    :color: primary
    :outline:

    :octicon:`mark-github` View on GitHub

A Dissect module implementing parsers for full volume encryption implementations,

> currently Microsoft's Bitlocker Disk Encryption (BDE) and Linux Unified Key Setup (LUKS1 and LUKS2).

Reviewer: ```suggestion
currently Microsoft's BitLocker Disk Encryption (BDE) and Linux Unified Key Setup (LUKS1 and LUKS2).
```

Tools
-----

fve-dd
~~~~~~

The ``fve-dd`` tool is used to decrypt Microsoft BitLocker (BDE) or Linux Unified Key Setup (LUKS1 and LUKS2) volumes
and write the decrypted content to a file.

If the input file is a disk with multiple volumes/partitions, the output file will be a disk image with the same partition layout,

> but with the encrypted volumes replaced by their decrypted content.

Reviewer: ```suggestion
with the encrypted volumes replaced by their decrypted content.
```

    :description: Utility to decrypt BitLocker or LUKS volumes and write them to a file.
    :hook:

Examples
^^^^^^^^

**BitLocker or LUKS volumes with passphrase:**

.. code-block:: console

    $ fve-dd encrypted.dd -p "mypassphrase" -o decrypted_volume.dd

**BitLocker volumes with recovery password:**

.. code-block:: console

    $ fve-dd encrypted.dd -r "123456-789012-345678-901234-567890-123456-789012-345678" -o decrypted.dd

**BitLocker volumes with .BEK file:**

BitLocker External Key (BEK) files can be used to unlock BitLocker encrypted volumes.
These files are typically stored on removable media like USB drives.

.. code-block:: console

    $ fve-dd encrypted.dd -f /path/to/recovery_key.BEK -o decrypted.dd

**LUKS volumes with key file:**

LUKS key files contain the encryption key and can be used instead of a passphrase.

.. code-block:: console

    $ fve-dd encrypted.dd -f /path/to/keyfile -o decrypted.dd

**LUKS volumes with specific key slot:**

You can specify which key slot to use when unlocking a LUKS volume:

.. code-block:: console

    $ fve-dd encrypted.dd -f /path/to/keyfile --key-slot 0 -o decrypted.dd

**LUKS volumes with key file offset and size:**

> For advanced use cases, you can specify the offset and size within a key file:

Reviewer: ```suggestion
Similar to ``cryptsetup`` utility, you can specify the offset and size within a key file:
```

**LUKS volumes with key file offset and size:**

> Similar to ``cryptsetup`` utility, you can specify the offset and size within a key file:

Reviewer: Forgot a 'the', my bad. After this we are good.
```suggestion
Similar to the ``cryptsetup`` utility, you can specify the offset and size within a key file:
```
